numbat/src/datetime.rs

```diff
@@ -1,82 +1,65 @@
-    tz_str.parse().ok()
+pub fn get_local_timezone_or_utc() -> TimeZone {
+    TimeZone::system()
 }
 
-#[cfg(feature = "local-timezone")]
-pub fn get_local_timezone_or_utc() -> Tz {
-    get_local_timezone().unwrap_or(chrono_tz::UTC)
-}
-
-#[cfg(not(feature = "local-timezone"))]
-pub fn get_local_timezone_or_utc() -> Tz {
-    chrono_tz::UTC
-}
-
-pub fn parse_datetime(input: &str) -> Option<DateTime<FixedOffset>> {
-    if let Ok(dt) = chrono::DateTime::parse_from_rfc3339(input) {
-        Some(dt)
-    } else if let Ok(dt) = chrono::DateTime::parse_from_rfc2822(input) {
+pub fn parse_datetime(input: &str) -> Option<Zoned> {
+    if let Ok(dt) = DateTime::strptime("%Y-%m-%dT%H:%M:%SZ", input) {
+        // RFC 3339
+        Some(dt.to_zoned(TimeZone::UTC).unwrap()) // TODO
+    } else if let Ok(dt) = rfc2822::parse(input) {
         Some(dt)
     } else {
         const FORMATS: [&str; 8] = [
             // 24 hour formats:
-            "%Y-%m-%d %H:%M:%S%.f",
-            "%Y/%m/%d %H:%M:%S%.f",
+            "%Y-%m-%d %H:%M:%S", // TODO: add support for fractional seconds
+            "%Y/%m/%d %H:%M:%S", // TODO: add support for fractional seconds
             "%Y-%m-%d %H:%M",
             "%Y/%m/%d %H:%M",
             // 12 hour formats:
-            "%Y-%m-%d %I:%M:%S%.f %p",
+            "%Y-%m-%d %I:%M:%S %p", // TODO: add support for fractional seconds
             "%Y-%m-%d %I:%M %p",
-            "%Y/%m/%d %I:%M:%S%.f %p",
+            "%Y/%m/%d %I:%M:%S %p", // TODO: add support for fractional seconds
             "%Y/%m/%d %I:%M %p",
         ];
 
         for format in FORMATS {
             // Try to match the given format plus an additional UTC offset (%z)
-            if let Ok(dt) = chrono::DateTime::parse_from_str(input, &format!("{format} %z")) {
+            if let Ok(dt) = Zoned::strptime(&format!("{format} %z"), input) {
                 return Some(dt);
             }
 
             // Try to match the given format plus an additional timezone name (%Z).
-            // chrono does not support %Z, so we implement this ourselves. We were
+            // jiff does not support %Z, so we implement this ourselves. We were
             // warned by developers before us not to write timezone-related code on
             // our own, so we're probably going to regret this.
 
             // Get the last space-separated word in the input string, and try to parse it
             // as a timezone specifier, then try to match the rest of the string with the
             // given format.
             if let Some((rest, potential_timezone_name)) = input.rsplit_once(' ') {
-                if let Ok(tz) = potential_timezone_name.parse::<Tz>() {
-                    if let Ok(ndt) = chrono::NaiveDateTime::parse_from_str(rest, format) {
-                        if let LocalResult::Single(dt) = ndt.and_local_timezone(tz) {
-                            return Some(dt.fixed_offset());
-                        }
```

Comment:
This is certainly nicer (easier to understand) using jiff. I remember that it took me a lot of time to get this right using chrono.